views/default/core/friends/collection.php

```diff
@@ -1,67 +1,55 @@
 <?php
+
 /**
  * View a friends collection
  *
  * @package Elgg
  * @subpackage Core
  *
- * @uses $vars['collection'] The individual friends collection
+ * @uses $vars['collection'] The access collection
  */
-
-$coll = $vars['collection'];
-
-if (is_array($vars['collection']->members)) {
-	$count = sizeof($vars['collection']->members);
-} else {
-	$count = 0;
+$collection = elgg_extract('collection', $vars);
+if (empty($collection)) {
+	return;
 }
 
-echo "<li><h2>";
+$members = get_members_of_access_collection($collection->id) ? : array();
```

Comment:
If we're re-writing this we should get away from outputting *all* friends.  These are trouble spots for scalability that we're constantly having to override in our projects become someone thinks it's cool to friend 10k people.